From 552afdbdad92e08d855edba662b6b36c988639b1 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Sun, 5 May 2024 23:19:32 +0200
Subject: [PATCH 0836/1695] plat-stm32mp1: let DTB register TRNG

Set CFG_WITH_TRNG upon CFG_STM32_RNG in platform stm32mp1. A TRNG
device will be registered if the DTB includes an enabled compatible node.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 062b5424bc47cc4b845e38cb7ebc689346a66db3 plat-stm32mp1: let DTB register TRNG]
Change-Id: Ie93c131b6b019b23459d15f9fdc7c99116d998ba
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/378334
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index b9e2dc8b5..d2ffb266d 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -301,6 +301,8 @@ ifneq ($(CFG_WITH_SOFTWARE_PRNG),y)
 $(call force,CFG_STM32_RNG,y,Required by HW RNG when CFG_WITH_SOFTWARE_PRNG=n)
 endif
 
+CFG_WITH_TRNG ?= $(CFG_STM32_RNG)
+
 ifeq ($(CFG_STM32_ETZPC),y)
 $(call force,CFG_STM32_FIREWALL,y)
 endif
-- 
2.47.2

